tests/MongoDB.Driver.Tests/Linq/Translators/PredicateTranslatorTests.cs

```diff
@@ -120,6 +120,118 @@ public void Any_with_a_predicate_on_documents()
                 "{ G : { $elemMatch : { D : \"Don't\", 'E.F' : 33 } } }");
         }
 
+        [Fact]
+        public void Any_with_a_predicate_on_document_itself()
+        {
+            Assert(
+                x => x.G.Any(g => g != null),
+                2,
+                "{ 'G' : { '$elemMatch' : { '$ne' : null } } }");
+
+            Assert(
+                x => x.G.Any(g => null != g),
+                2,
+                "{ 'G' : { '$elemMatch' : { '$ne' : null } } }");
+
+            Assert(
+                x => x.G.Any(g => g.E.I.Any(i => i == "insecure")),
+                1,
+                "{ \"G.E.I\" : { '$elemMatch' : { '$eq': 'insecure' } } }");
+
+            // todo: the generated query is wrong: {G.E.I= { "$elemMatch" : { "$or" : [{ "" : "insecure" }, { "" : "igloo" }] } }}.
+            //Assert(
+            //    x => x.G.Any(g => g.E.I.Any(i => i == "insecure" || i == "igloo")),
+            //    ,
+            //    "");
+
+            // todo: the generated query is wrong: {G= { "$elemMatch" : { "" : { "$ne" : null }, "E" : { "$ne" : null } } }}.
+            //Assert(
+            //    x => x.G.Any(g => g != null && g.E != null),
+            //    ,
+            //    "");
+        }
+
+        [Fact]
+        public void Any_with_a_predicate_on_document_itself_and_objectId()
+        {
+            Assert(
+                x => x.G.Any(g => g.Ids.Any(i => i == ObjectId.Parse("111111111111111111111111"))),
+                1,
+                "{ 'G.Ids' : { '$elemMatch' : { '$eq' : ObjectId('111111111111111111111111') } } }");
+
+        }
+
+        [Fact]
+        public void Any_with_a_predicate_on_documents_itself_and_ClassEquals()
+        {
+            var c1 = new C()
+            {
+                D = "Dolphin",
+                E = new E()
+                {
+                    F = 55,
+                    H = 66,
+                    I = new List<string>()
+                    {
+                        "insecure"
+                    }
+                }
+            };
+            Assert(
+                x => x.G.Any(g => g == c1),
+                1,
+                "{ \"G\" : { \"$elemMatch\" : { \"Ids\" : null, \"D\" : \"Dolphin\", \"E\" : { \"F\" : 55, \"H\" : 66, \"I\" : [\"insecure\"] }, \"S\" : null, \"X\" : null } } }");
+
+            // todo: the generated query is wrong
```

Comment:
Same as above... put in JIRA ticket instead of as comments here?

Reply:
I will do it